ouroboros-consensus/src/Ouroboros/Consensus/Storage/ImmutableDB/Chunks/Layout.hs

```diff
@@ -204,6 +210,21 @@ chunkSlotForBlockOrEBB ci = \case
     Block slot  -> chunkSlotForRegularBlock  ci slot
     EBB   epoch -> chunkSlotForBoundaryBlock ci epoch
 
+-- | Chunk slot for 'Tip'
+chunkSlotForTip :: ChunkInfo -> Tip blk -> ChunkSlot
+chunkSlotForTip ci Tip { tipSlotNo, tipIsEBB } = case tipIsEBB of
+    IsNotEBB -> chunkSlotForRegularBlock ci tipSlotNo
+    IsEBB    -> assertChunkCanContainEBB chunkIndex relativeSlotChunkSize $
+                 UnsafeChunkSlot chunkIndex $ firstBlockOrEBB ci chunkIndex
```

Comment:
Translate to `BlockOrEBB` and then call `chunkSlotForBlockOrEBB`

Reply:
Hmm, if I try this, it doesn't really look better than what I have: for the conversion from `SlotNo` to `EpochNo` I need `chunkSlotForRegularBlock`.